src/XrdCrypto/XrdCryptosslAux.cc

```diff
@@ -730,16 +730,26 @@ time_t XrdCryptosslASN1toUTC(const ASN1_TIME *tsn1)
        &(ltm.tm_year), &(ltm.tm_mon), &(ltm.tm_mday),
        &(ltm.tm_hour), &(ltm.tm_min), &(ltm.tm_sec),
                                       &zz) != 7) || (zz != 'Z')) {
-       return -1;
+       // Try GeneralizedTime
+       if ((sscanf((const char *)(tsn1->data),
+           "%04d%02d%02d%02d%02d%02d%c", 
+           &(ltm.tm_year), &(ltm.tm_mon), &(ltm.tm_mday),
+           &(ltm.tm_hour), &(ltm.tm_min), &(ltm.tm_sec),
+                                          &zz) != 7) || (zz != 'Z')) {
+           return -1;
+       }
    }
    // Init also the ones not used by mktime
    ltm.tm_wday  = 0;        // day of the week 
    ltm.tm_yday  = 0;        // day in the year
    ltm.tm_isdst = 0;        // we will correct with an offset without dst
    //
    // Renormalize some values: year should be modulo 1900
-   if (ltm.tm_year < 90)
+   if (ltm.tm_year < 50)
       ltm.tm_year += 100;
+   // In case GeneralizedTime is used
+   if (ltm.tm_year > 2000)
+      ltm.tm_year -= 1900;
```

Comment:
I think we need `ltm.tm_year > 1900` here instead of `ltm.tm_year > 2000`, otherwise I see wrong dates (i.e. with year >3600) being generated.

Reply:
In the end, I think this is what would give the most reasonable answers:

```cpp
   if (ltm.tm_year < 50)
      ltm.tm_year += 2000;
   else if (ltm.tm_year < 100)
      ltm.tm_year += 1900;

   ltm.tm_year -= 1900;
```
This works for years 00(2000) 01(2001) 49(2049) 51(1951) 1800 1900 1949 1951 2000, and 2100 without wrapping into 3000+.